src/main/java/world/bentobox/bentobox/managers/BlueprintsManager.java

```diff
@@ -116,29 +139,71 @@ private File getBlueprintsFolder(@NonNull GameModeAddon addon) {
      * @param addon the {@link GameModeAddon} to load the blueprints of.
      */
     public void loadBlueprintBundles(@NonNull GameModeAddon addon) {
-        blueprintBundles.putIfAbsent(addon, new ArrayList<>());
+        blueprintBundles.put(addon, new HashMap<>());
         File bpf = getBlueprintsFolder(addon);
         for (File file: Objects.requireNonNull(bpf.listFiles((dir, name) ->  name.toLowerCase(Locale.ENGLISH).endsWith(BLUEPRINT_BUNDLE_SUFFIX)))) {
             try {
-                blueprintBundles.get(addon).add(gson.fromJson(new FileReader(file), BlueprintBundle.class));
+                BlueprintBundle bb = gson.fromJson(new FileReader(file), BlueprintBundle.class);
+                blueprintBundles.get(addon).put(bb.getUniqueId(), bb);
+                plugin.log("Loaded Blueprint Bundle '" + bb.getUniqueId() + "' for " + addon.getDescription().getName());
             } catch (Exception e) {
                 plugin.logError("Could not load blueprint bundle " + file.getName() + " " + e.getMessage());
+                e.printStackTrace();
             }
         }
+        if (blueprintBundles.get(addon).isEmpty()) {
+            makeDefaults(addon);
+            loadBlueprintBundles(addon);
+        }
+        // Load blueprints
+        loadBlueprints(addon);
+    }
+
+    /**
+     * This should never be needed and is just a boot strap
+     * @param addon
+     */
+    private void makeDefaults(@NonNull GameModeAddon addon) {
```

Comment:
This method is a useful one to study. It is used to make default blueprint bundle and a single block blueprint in the event there are no files in the Addon's blueprint folder. Generally BlueprintManager is where you can manage bundles and blueprints.